hail/python/hailtop/batch/contrib/regenie/Dockerfile

```diff
@@ -0,0 +1,36 @@
+# Adapted by Hail Team from Regenie
+# Original attribution line:
+# License path: regenie/LICENSE
+# Thanks to Nathan Weeks for sharing this Dockerfile
+# minor edits were made
+# Filename: Dockerfile
+
+FROM ubuntu:16.04 AS builder
+
+WORKDIR /src
+
+ADD http://code.enkre.net/bgen/tarball/release/v1.1.7 v1.1.7.tgz
```

Comment:
Why do they recommend against it?

Reply:
Since we need to maintain the source files anyway, my plan is to avoid maintaining our own, very similar docker file, and just add the regenie repo as a submodule. We can build using their docker file, and pin to a given commit. There will be no need to check for docker file changes in between commit pins. Would this work for you?

Workflow for batch is something like

git clone --recursive https://github.com/hail-is/hail
# do this step only if we update the submodule
docker build hail/python/hailtop/batch/genetics/regenie/regenie -t gcr.io/hailgenetics/...

Reply:
Then there is no need to worry about Dockerfile code style; we're clearly using someone else's docker file (the docker file of the project we're serving in batch), style is not our concern.

Reply:
Ugh, unfortunately Batch doesn't seem to allow me to modify how docker run works, which means that any ENTRYPOINT completely fubars Batch. I think we should fix that (override entrypoint), would you accept a PR?

So we need to go back to maintaining a lightly modified docker file. The remote url issue appears fixed using DOCKER_BUILDKIT=1, is the recommended solution by docker folks, and has touted build performance advantages: https://github.com/moby/moby/issues/15717#issuecomment-495426529 , https://docs.docker.com/develop/develop-images/build_enhancements/. Since the solution you provided isn't recommended, would you mind use of buildkit?

Reply:
I'm very interested in an entry point fix. It's a longstanding bug.